### PR TITLE
Implement updated Deneb builder REST API

### PR DESCRIPTION
**Pull request by StefanBratanov**
## PR Description

Implement the required REST API  changes as per https://github.com/ethereum/builder-specs/pull/90

- Changed `BuilderBid` to have blobKzgCommitments. Defaulted `getOptionalBlindedBlobsBundle()` to empty to avoid changing block production flow for now.
- Removed the notion of `SignedBlindedBlockContainer` passing in `getPayload`. Passing only `SignedBeaconBlock` now.

## Fixed Issue(s)
helps with builder block production flow for https://github.com/Consensys/teku/issues/7654

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/builder/BuilderBid.java [hunk 1]
```diff
@@ -17,13 +17,20 @@
...
+import tech.pegasys.teku.infrastructure.ssz.SszList;
 import tech.pegasys.teku.spec.datastructures.execution.ExecutionPayloadHeader;
+import tech.pegasys.teku.spec.datastructures.type.SszKZGCommitment;
 
 public interface BuilderBid extends SszContainer {
 
   ExecutionPayloadHeader getHeader();
 
-  Optional<BlindedBlobsBundle> getOptionalBlindedBlobsBundle();
+  // TODO: make sure removed after moving to sidecar inclusion proof
```
Comment (zilm13) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/builder/BuilderBid.java]: May use `@Deprecated` instead

File: ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/ExecutionBuilderModule.java [hunk 2]
```diff
@@ -378,7 +378,7 @@ private SafeFuture<BuilderPayload> getPayloadFromBuilder(
             () ->
                 new RuntimeException(
                     "Unable to get payload from builder: builder endpoint not available"))
-        .getPayload(signedBlindedBlockContainer)
+        .getPayload(signedBlindedBlockContainer.getSignedBlock())
```
Comment (zilm13) [inline ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/ExecutionBuilderModule.java]: As I understand we will need start from block higher in call flow, you are just cutting changes here for the moment. I'd make at least input of this function with Block, and update logging line, just at least to make it correct

**Review by zilm13 — APPROVED** [review]
LGTM

File: ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/ExecutionBuilderModule.java (hunk 2 above)
Comment (StefanBratanov) [reply ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/ExecutionBuilderModule.java]: good catch, I like it.

**Pull request merged**